projects/bird-atlas/src/app/app.component.ts

```diff
@@ -25,7 +34,22 @@ import { TranslateService } from '@ngx-translate/core';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class AppComponent {
-  constructor(translate: TranslateService) {
+  breadcrumbs$: Observable<IBreadcrumb[]> = this.breadcrumbs.breadcrumbs$.pipe(tap(() => this.cdr.detectChanges()));
```

Comment:
I'm just wondering if `this.cdr.detectChanges()` is needed here. Shouldn't the changes be always detected when using async pipes?

Reply:
Yeah I don't know why this one is here, seems to work without it. There's a similar case here: https://github.com/luomus/laji/blob/bd8011c53fa90833f32fe7a20bde6da144c0b073/projects/bird-atlas/src/app/+species/species-info/species-info.component.ts#L53, but in this case the detectChanges was actually necessary and I have no idea why. This detectChanges fixes a bug where the spinner would not disappear after navigating to a species page -> another page -> back to the same species page.

Reply:
That case in `species-info.component`is weird but we have had similar cases before... The problem is that when navigating back to species page, the observable emits a value too soon (as all the values are in cache). I'm not sure exactly but I guess that causes `isLoading` changing in the middle of Angular change detection cycle and that's something that shouldn't happen. We have usually fixed this kind of problems by adding `delay(0)` before the final `tap` instead of `this.cdr.detectChanges`